ui/logviewer/App.jsx

```diff
@@ -214,6 +223,23 @@ class App extends React.PureComponent {
     }
   };
 
+  logFormatter = (log) => {
+    if (/INFO - TEST-(OK)|(PASS)/.test(log))
+      return <span style={{ color: 'green' }}>{log}</span>;
+    if (/INFO - TEST-START/.test(log))
+      return <span style={{ color: 'blue' }}>{log}</span>;
+    if (/INFO - TEST-/.test(log))
+      return <span style={{ color: 'yellow' }}>{log}</span>;
+    if (/!!!/.test(log)) return <span style={{ color: 'fuchsia' }}>{log}</span>;
+    if (/Browser Chrome Test Summary$/.test(log))
+      return <span style={{ color: 'white' }}>{log}</span>;
+    if (/((INFO -)|([\s]+))(Passed|Failed|Todo):/.test(log))
+      return <span style={{ color: 'white' }}>{log}</span>;
+    if (/INFO/.test(log))
+      return <span style={{ color: '#5b7f7f' }}>{log}</span>;
```

Comment:
Since you're repeating the `<span>` element multiple times, you can reduce duplication by returning the span only once but assigning a new value to a `color` variable, ex:
```
        if (/INFO - TEST-(OK)|(PASS)/.test(log))
         color = 'green'
```